diff --git a/patches/originals/bosta-woocommerce.php b/plugins/bosta-woocommerce/bosta-woocommerce.php
index 0de39e2..6524e69 100644
--- a/patches/originals/bosta-woocommerce.php
+++ b/plugins/bosta-woocommerce/bosta-woocommerce.php
@@ -8 +8 @@
- * Version: 4.5.6
+ * Version: 4.5.7
@@ -129 +129 @@ const BOSTA_ENV_URL_V2 = 'https://app.bosta.co/api/v2';
-const PLUGIN_VERSION = '4.5.6';
+const PLUGIN_VERSION = '4.5.7';
@@ -139 +139 @@ function bosta_send_api_request($method, $url, $APIKey = null, $body = null)
-		'timeout' => 30,
+		'timeout' => 300,
@@ -332,0 +333,4 @@ function bosta_get_city_areas()
+				// filter not covered districts
+				if (empty($district['dropOffAvailability'])) {
+					continue;
+				}
@@ -737,0 +742,5 @@ function bosta_add_dynamic_area_dropdown_to_checkout($fields)
+		// Set the selected area from session if available, otherwise default to empty string (php part)
+		$selected = WC()->session
+			? WC()->session->get('billing_area')
+			: '';
@@ -743 +752 @@ function bosta_add_dynamic_area_dropdown_to_checkout($fields)
-				'' => __('Select an option...', 'woocommerce'),
+				'' => __('حدد خيارا...', 'woocommerce'),
@@ -747,0 +757,4 @@ function bosta_add_dynamic_area_dropdown_to_checkout($fields)
+			'custom_attributes' => array(
+				'data-selected' => $selected,
+			),
+			'class' => array('update_totals_on_change'), // Triggers updated_checkout when Area changes.
@@ -754 +767,8 @@ function bosta_add_dynamic_area_dropdown_to_checkout($fields)
-            var select2_args = { minimumResultsForSearch: 5 };
+            var select2_args = {
+                minimumResultsForSearch: 5,
+                language: {
+                       noResults: function () {
+                       return 'لم يتم العثور على نتائج';
+                    }
+                }
+            };
@@ -760,2 +780,5 @@ function bosta_add_dynamic_area_dropdown_to_checkout($fields)
-        $('select.wc-enhanced-select').val('').trigger('change');
-		$('#billing_state').val('').trigger('change');
+		/* Fix not selecting the area dropdown on page reload if the state is already selected and has a saved area value */
+        // $('select.wc-enhanced-select').val('').trigger('change');
+		// $('#billing_state').val('').trigger('change');
+		/* populate area dropdown using saved state */
+    	$('#billing_state').trigger('change');
@@ -792 +815 @@ function bosta_add_area_dropdown_admin_order($order)
-			'<label for="billing_area">' . __('Area', 'woocommerce') . '</label>' .
+			'<label for="billing_area">' . __('الحي', 'woocommerce') . '</label>' .
@@ -830,0 +854,2 @@ function bosta_enqueue_dynamic_area_dropdown_script()
+						// Get the custom attr value that was set by php (data-selected) to re-select it after repopulating the dropdown, if it exists
+						var savedArea = areaDropdown.val() || areaDropdown.attr('data-selected');
@@ -841,0 +867,4 @@ function bosta_enqueue_dynamic_area_dropdown_script()
+						if (savedArea) {
+							areaDropdown.val(savedArea);
+						}
