diff --git a/plugins/bosta-woocommerce/bosta-woocommerce.original.php b/plugins/bosta-woocommerce/bosta-woocommerce.php
index 0de39e2..8c9ef20 100644
--- a/plugins/bosta-woocommerce/bosta-woocommerce.original.php
+++ b/plugins/bosta-woocommerce/bosta-woocommerce.php
@@ -330,6 +330,12 @@ function bosta_get_city_areas()
 			$city_code = $city['cityCode'];
 			$city_areas = '';
 			foreach ($city['districts'] as $district) {
+
+				// 🔥 فلترة حسب التغطية
+				if (empty($district['dropOffAvailability'])) {
+					continue;
+				}
+
 				$zone_name = $is_arabic ? $district['zoneOtherName'] : $district['zoneName'];
 				$district_name = $is_arabic ? $district['districtOtherName'] : $district['districtName'];
 
