diff --git a/plugins/bosta-woocommerce/bosta-woocommerce.php b/plugins/bosta-woocommerce/bosta-woocommerce.php
index 0de39e2..2f8de31 100644
--- a/plugins/bosta-woocommerce/bosta-woocommerce.php
+++ b/plugins/bosta-woocommerce/bosta-woocommerce.php
@@ -330,6 +330,10 @@ function bosta_get_city_areas()
 			$city_code = $city['cityCode'];
 			$city_areas = '';
 			foreach ($city['districts'] as $district) {
+				// filter not covered districts
+				if (empty($district['dropOffAvailability'])) {
+					continue;
+				}
 				$zone_name = $is_arabic ? $district['zoneOtherName'] : $district['zoneName'];
 				$district_name = $is_arabic ? $district['districtOtherName'] : $district['districtName'];
 
@@ -751,7 +755,14 @@ function bosta_add_dynamic_area_dropdown_to_checkout($fields)
 		wc_enqueue_js("
     jQuery(document).ready(function($) {
         $(':input.wc-enhanced-select').filter(':not(.enhanced)').each(function() {
-            var select2_args = { minimumResultsForSearch: 5 };
+            var select2_args = {
+                minimumResultsForSearch: 5,
+                language: {
+                       noResults: function () {
+                       return 'لم يتم العثور على نتائج';
+                    }
+                }
+            };
             $(this).select2(select2_args).addClass('enhanced');
         });
 
